### PR TITLE
Improved TV Shows view to mimic iTunes

### PR DESCRIPTION
**Pull request by delut**
I've changed quite a bit of things here to make it more similar to iTunes. Here are some screenshots:

screen shot 2016-04-03 at 19 21 15
screen shot 2016-04-03 at 19 21 27
screen shot 2016-04-03 at 19 21 37

It works in the following way:
1. The Show view has been deleted. If a user enters a show, it will be taken to the latest season. So if you tap the "Breaking Bad" show in Popular it will take you to the 5th season (shown in screenshot 1).
2. Then the user can press the seasons button and will prompt him to choose the desired season (Season Picker in screenshot 3).
3. All the episodes for the season are listed in the carousel shown in screenshot 2, with a description beneath. Tapping on a thumbnail in the carousel will show the quality picker introduced in 0.4.1.

Let me know what you think, or if I can change something to make it better.

### CODE REVIEW

**Comment by drew5494** [comment]
Looks sexy

File: PopcornTime/Managers/WatchlistManager.swift [hunk 1]
```diff
@@ -172,16 +172,26 @@ public class WatchlistManager {
...
+            completion?(json: nil)
         }
     }
 
     func writeJSONFile(json: [[String : AnyObject]]) {
         do {
-            print(json)
             let json = try NSJSONSerialization.dataWithJSONObject(json, options: .PrettyPrinted)
-            try json.writeToFile(self.jsonFilePath, options: .AtomicWrite)
+            NSUserDefaults.standardUserDefaults().setObject(json, forKey: "watchlist")
```
Comment (buzzel) [inline PopcornTime/Managers/WatchlistManager.swift]: There was an issue with the watchlist on the device, does this fix it?

File: PopcornTime/Managers/WatchlistManager.swift [hunk 2]
```diff
@@ -71,7 +71,7 @@ public struct WatchItem {
 public class WatchlistManager {
 
     private var jsonFilePath: String! {
-        let paths = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)
+        let paths = NSSearchPathForDirectoriesInDomains(NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomainMask.UserDomainMask, true)
```
Comment (buzzel) [inline PopcornTime/Managers/WatchlistManager.swift]: In my opinion the shorter syntax looks better, but thats just me nitpicking.

**Comment by buzzel** [comment]
@delut If you wouldnt mind running Swiftlint on the project to clean up any of the lint errors, I can merge this in. `brew install swiftlint` `swiftlint autocorrect` will fix most of the issues. If you run `swiftlint` it will tell you what its found.

**Comment by delut** [comment]
Hi @buzzel I ran the autocorrect, and its now showing me only one issue:

`Linting 'ActionHandler.swift' (1/22)
/Users/Tomi/Documents/PopcornTimeTV/PopcornTime/ActionHandler.swift:95:5: warning: Function Body Length Violation: Function body should span 40 lines or less excluding comments and whitespace: currently spans 41 lines (function_body_length)`

What should I do? Thanks.

**Comment by buzzel** [comment]
@delut It just means that its a huge block of code https://github.com/delut/PopcornTimeTV/blob/master/PopcornTime/ActionHandler.swift#L95. I'm not worries about those, unless you had ideas on cleaning it up? If not, no worries though, just add `// swiftlint:disable this function_body_length` and run the linter again or add it to `.swiftlint.yml`

**Comment by delut** [comment]
I added the `function_body_length` parameter to the `.swiftlint.yml` but the houndci-bot is still complaining. It is also pointing out some other things that didn't appear in the terminal when I ran the `swiftlint` command.

**Comment by buzzel** [comment]
Yeah, Im not sure what that dogs problem is haha. Looks good to merge to me though.

**Pull request merged**